pkg/api/types.go

```diff
@@ -448,6 +448,8 @@ const (
 // Host path volumes do not support ownership management or SELinux relabeling.
 type HostPathVolumeSource struct {
 	Path string `json:"path"`
+	// Mount the host path with propagation shared. Docker only.
+	Shared bool `json:"shared"`
```

Comment:
Hmm, I thought we all agreed that:
 - `private` should be the default
 - `shared` should be togglable for users that know what they're doing, and only for `hostPath`.

This way, we're not exposing very much Docker-specific (atm) semantics, only a high-low switch which is on or off. 

@thockin thought that a propagnation enum is too detailed, especially since this is v1 api already. (and I agree)

By letting the user gain the highest priv possible (with shared), they can do whatever they have to.

BTW, is there any difference between shared and rshared?